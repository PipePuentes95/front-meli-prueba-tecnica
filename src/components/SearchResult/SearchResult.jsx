import React, { useState, useEffect } from 'react'
import { useSearchParams } from "react-router-dom";
import axios from 'axios';

import Breadcrumb from '../Breadcrumb/Breadcrumb';
import Loader from '../Loader/Loader';
import ProductCard from '../ProductCard/ProductCard';

import './SearchResult.scss';

const SearchResult = () => {

  const [searchParams] = useSearchParams();

  const [items, setItems] = useState([]);
  const [categories, setCategories] = useState([]);
  const [withoutResults, setWithoutResults] = useState(false);

  const getSearchResult = async (params) => {
    setItems([]);
    setWithoutResults(false);

    const { data } = await axios.get(`http://localhost:4001/api/items?q=${ params }`);

    const categories = data.categories.filter((_, index) => index < 5);

    if(data.length === 0) return setWithoutResults(true);
    setWithoutResults(false);
    setItems(data.items);
    setCategories(categories);
  }

  useEffect(()=> {
    const params = searchParams.get('q');
    getSearchResult(params);
  }, [searchParams]);

  return (
    <div className="search-result">
      { (items.length !== 0 && !withoutResults) && (
        <div className='container'>
          <Breadcrumb categories={ categories } />
            <div className='products'>
              <div className="products__wrapper">
                { items.map((item) => (
                  <ProductCard
                    key={ item.id }
                    id={ item.id }
                    title={ item.title }
                    price={ item.price }
                    picture={ item.picture }
                    free_shipping={ item.free_shipping }
                    state_name={ item.state_name } />
                )) }
              </div>
            </div>
        </div>
      )}

      { (items.length === 0 && !withoutResults) &&(
        <div className='container'>
          <Loader />
        </div>
      )}

      { (items.length === 0 && withoutResults) &&(
        <div className='container'>
          <div className="without-results">
            <h3>No se encontraron resultados de busqueda</h3>
          </div>
        </div>
      )}
    </div>
  );
};

export default SearchResult;
