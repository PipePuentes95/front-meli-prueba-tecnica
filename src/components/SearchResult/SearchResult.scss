@import '../../styles/variables';

.search-result {
  min-height: calc(100vh - 60px);

  padding-bottom: 16px;

  background-color: $grey-4;

  .products {
    @media(max-width: 764px) {
      width: 100%;
    }
  
    width: 80%;
  
    margin: 0 auto;

    &__wrapper {
      background-color: $white;

      border-radius: 4px;
      overflow: hidden;
    }
  }
}

.without-results {
  @media(max-width: 764px) {
    width: 100%;
  }

  width: 80%;
  height: 400px;

  display: flex;
  align-items: center;
  justify-content: center;

  margin: 0 auto;
  padding: 10px 0;

  h3 {
    font-size: 50px;
    color: $grey-2;
  }
}