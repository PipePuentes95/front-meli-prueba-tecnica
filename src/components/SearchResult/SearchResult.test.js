import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';

import SearchResult from './SearchResult';

it('should take a snapshot', () => {
  const pathname = createMemoryHistory();
  const { asFragment } = render (
    <Router location={ pathname.location }>
      <SearchResult />
    </Router>
  )

  expect(asFragment(<SearchResult />)).toMatchSnapshot();
});