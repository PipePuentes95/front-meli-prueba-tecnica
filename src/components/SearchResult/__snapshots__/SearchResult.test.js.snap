// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="search-result"
  >
    <div
      class="container"
    >
      <div
        class="loader__wrapper"
      >
        <h3>
           Cargando ...
        </h3>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
